spec/presenters/coronavirus/sub_section_json_presenter_spec.rb

```diff
@@ -94,157 +86,6 @@
-      expect(sub_section_hash[:list].first[:label]).to eq(label)
-      expect(sub_section_hash[:list].first[:url]).to eq(path)
-    end
-
-    it "has a null title" do
-      expect(sub_section_hash.keys).to include(:title)
-      expect(sub_section_hash[:title]).to be_nil
-    end
-
-    context "with a title" do
-      let(:title) { Faker::Lorem.sentence }
-      let(:title_markup) { "## #{title}" }
-      let(:group) { [title_markup, link] }
-
-      it "puts the link path and label into list" do
-        expect(sub_section_hash[:list].first[:label]).to eq(label)
-        expect(sub_section_hash[:list].first[:url]).to eq(path)
-      end
-
-      it "has the title" do
-        expect(sub_section_hash[:title]).to eq(title)
-      end
-    end
-
-    context "with a spaceless title" do
-      let(:title) { Faker::Lorem.sentence }
-      let(:title_markup) { "###{title}" }
-      let(:group) { [title_markup, link] }
-
-      it "has the title" do
-        expect(sub_section_hash[:title]).to eq(title)
-      end
-    end
-
-    context "with a full url in link" do
-      let(:url) { Faker::Internet.url }
-      let(:link) { "[#{label}](#{url})" }
-
-      it "puts the link path and label into list" do
-        expect(sub_section_hash[:list].first[:label]).to eq(label)
-        expect(sub_section_hash[:list].first[:url]).to eq(url)
-      end
-    end
-
-    context "with a full secure url in link" do
-      let(:url) { Faker::Internet.url }
-      let(:link) { "[#{label}](#{url})" }
-
-      it "puts the link path and label into list" do
-        expect(sub_section_hash[:list].first[:label]).to eq(label)
-        expect(sub_section_hash[:list].first[:url]).to eq(url)
-      end
-    end
-
-    context "with spaces" do
-      let(:link) { " [ #{label} ] ( #{path} ) " }
-
-      it "puts the link path and label into list" do
-        expect(sub_section_hash[:list].first[:label]).to eq(label)
-        expect(sub_section_hash[:list].first[:url]).to eq(path)
-      end
-    end
-
-    context "with two links" do
-      let(:label_two) { Faker::Lorem.sentence }
-      let(:path_two)  { "/#{File.join(Faker::Lorem.words)}" }
-      let(:link_two) { "[#{label_two}](#{path_two})" }
-      let(:group) { [link, link_two] }
-
-      it "puts the link path and label into list" do
-        expect(sub_section_hash[:list].first[:label]).to eq(label)
-        expect(sub_section_hash[:list].first[:url]).to eq(path)
-      end
-
-      it "puts the second link path and label into list" do
-        expect(sub_section_hash[:list].last[:label]).to eq(label_two)
-        expect(sub_section_hash[:list].last[:url]).to eq(path_two)
-      end
-    end
-
-    context "with unknown content" do
-      let(:group) { %w[unknown] }
-
-      it "adds an error" do
-        expect { sub_section_hash }.to raise_error(Coronavirus::SubSectionJsonPresenter::MarkdownInvalidError)
-      end
-    end
-
-    context "when link is to a subtopic path" do
```

Comment:
I think these tests should be put back as they test that links to a hub page are being set as "featured".

Reply:
Sorry, just so I understand, hub pages are meant to be featured whether they're listed as featured or not?

Reply:
Yes. It's a bit confusing I know, but the code explicitly sets links to hub pages as featured links: https://github.com/alphagov/collections-publisher/blob/9e8fe1f11407d0191a14170b6ee445d90832a4d6/app/presenters/coronavirus/sub_section_json_presenter.rb#L96-L98

Reply:
Okay cool, in that case, I'll add in a test for `build_link` to do this because that's where it's actually being done and that for some reason wasn't tested directly before.